backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/progressbar/BpkProgressBar.kt

```diff
@@ -0,0 +1,141 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.LinearProgressIndicator
+import androidx.compose.material3.ProgressIndicatorDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import androidx.compose.ui.unit.Dp
+import net.skyscanner.backpack.compose.annotation.BpkPreviews
+
+enum class BpkProgressBarSize {
+    Small,
+    Large,
+}
+
+@Composable
+fun BpkProgressBar(
+    value: Float,
+    // accessibilityLabel: String, <- not sure if we need this, in its current state talkback reads ("zero percent progress bar")
+    modifier: Modifier = Modifier,
+    size: BpkProgressBarSize = BpkProgressBarSize.Small,
+    min: Int = 0,
+    max: Int = 1,
+    stepped: Boolean = false,
+) {
+    val height = if (size == BpkProgressBarSize.Small) 8.dp else 16.dp
+
+    val progress = value
+        .coerceIn(min.toFloat(), max.toFloat())
+        .div(max.toFloat())
+
+    val animatedProgress by animateFloatAsState(
+        targetValue = progress,
+        animationSpec = ProgressIndicatorDefaults.ProgressAnimationSpec,
+    )
+
+    Box(modifier = modifier) {
+        LinearProgressIndicator(
+            progress = animatedProgress,
+            modifier = Modifier
+                .height(height)
+                .fillMaxWidth()
+                .clip(CircleShape),
+            color = BpkTheme.colors.coreAccent,
+            trackColor = BpkTheme.colors.surfaceHighlight,
+            strokeCap = if (stepped) StrokeCap.Butt else StrokeCap.Round,
+        )
+
+        if (stepped) {
+            val dividers = max - 1
+
+            if (dividers > 0) {
+                BuildDividers(
+                    height = height,
+                    dividers = dividers,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun BuildDividers(
+    height: Dp,
+    dividers: Int,
+) {
+    Row(
+        modifier = Modifier
+            .height(height)
+            .fillMaxWidth(),
+        horizontalArrangement = Arrangement.SpaceEvenly,
+        content = {
+            repeat(dividers) {
+                Box(
+                    modifier = Modifier
+                        .fillMaxHeight()
+                        .width(2.dp)
+                        .background(BpkTheme.colors.surfaceDefault),
+                )
+            }
+        },
+    )
+}
+
+@BpkPreviews
+@Composable
+internal fun PreviewBpkProgressBar() {
```

Comment:
We usually create the previews inside the app module, and then re-use it for tests and stories

Reply:
Ahh, I understand now. Removed 🧹